test/spec/groupCtrlSpec.js

```diff
@@ -65,3 +65,31 @@ describe('Groups Controller', function() {
     expect(myGuilds).to.be.called
   });
 });
+
+describe("Autocomplete controller", function() {
```

Comment:
I think the problem here is that the RootCtrl wasn't getting loaded in. The tests themselves work brilliantly.

Here's my refactored version:

```
describe("Autocomplete controller", function() {
  var scope, ctrl, user, $rootScope;

  beforeEach(function() {
    module(function($provide) {
      $provide.value('User', {});
    });

    inject(function($rootScope, $controller){
      user = specHelper.newUser();
      user._id = "unique-user-id";

      scope = $rootScope.$new();

      // Load RootCtrl to ensure shared behaviors are loaded
      $controller('RootCtrl',  {$scope: scope, User: {user: user}});

      ctrl = $controller('AutocompleteCtrl', {$scope: scope});
    });
  });

  it('filtering with undefined query (not loaded yet) defaults to true', function() {
      expect(scope.filterUser({user: "boo"})).to.be.eq(true);
  })

  it('filtering with null query (no typing yet) defaults to true', function() {
    scope.query = null
    expect(scope.filterUser({user: "boo"})).to.be.ok
  })

  it('filtering with prefix element will return true', function() {
    scope.query = {text: "pre"}
    expect(scope.filterUser({user: "prefix"})).to.be.ok
  })

  it('filtering with nonprefix element will return false', function() {
    scope.query = {text: "noprefix"}
    expect(scope.filterUser({user: "prefix"})).to.not.be.ok
  })
});
```

The rootCtrl instantiation has to happen so many times, I'll probably pull it out into a helper method, but for now, instantiating it manually is fine.